lib/ts/recipe/thirdparty/components/library/providerButton.tsx

```diff
@@ -21,9 +23,37 @@ export type ProviderButtonProps = {
     logo?: JSX.Element;
 };
 
-export default function ProviderButton({ logo, providerName, displayName }: ProviderButtonProps) {
+export default function ProviderButton({ logo, providerName, displayName }: ProviderButtonProps): React.ReactElement {
     const t = useTranslation();
     const providerStyleName = `provider${providerName}`;
+    const buttonTextContainerRef = useRef<HTMLDivElement>(null);
+
+    const SCROLL_ANIMATION_CLASS = "scroll-text-animation";
+
+    const isTextOverflowing = (element: Element) => {
+        return element.scrollWidth > element.clientWidth;
+    };
```

Comment:
Please move this to the root scope so it's not re-created on each render (and so that the linter will not complain about it missing in the dependency array)

Reply:
Done